src/app/static/src/app/store/genericChart/actions.ts

```diff
@@ -0,0 +1,18 @@
+import {ActionContext, ActionTree} from 'vuex';
+import {RootState} from "../../root";
+import {api} from "../../apiService";
+import {GenericChartState} from "./genericChart";
+import {GenericChartMutation} from "./mutations";
+
+export interface MetadataActions {
+    getGenericChartMetadata: (store: ActionContext<GenericChartState, RootState>) => void
+}
+
+export const actions: ActionTree<GenericChartState, RootState> & MetadataActions = {
+    async getGenericChartMetadata(context) {
+        await api<GenericChartMutation, "">(context)
+            .withSuccess(GenericChartMutation.GenericChartMetadataFetched)
```

Comment:
Out of interest, is this the normal pattern for setting data: setting the fetched flag with a mutation, and then off the back of that mutation setting the data mutation? As opposed to setting the data directly off the action payload and then setting the fetched flag.

Reply:
>could you do a check that the chartConfigNode has certain expected properties before casting it and setting config; then account for the failure case.

We could be more defensive here but tbh since we should just be reading a resource file that should never change it seems a little over the top.

Reply:
>Out of interest, is this the normal pattern for setting data: setting the fetched flag with a mutation, and then off the back of that mutation setting the data mutation? As opposed to setting the data directly off the action payload and then setting the fetched flag.

Hm, this might be a misleadingly named mutation - it doesn't actually set a fetched flag - we don't have one for this metadata since its fetching is synchronous and invisible to the user. The mutation just sets the data, so the 'fetched' of the name is more 'this thing has happened' rather than 'this is what we're setting'. We have both styles, that latter usually start with `Set`. We should probably standardise on one or the other, but this is consistent with the style in the metadata module which this most closely resembles...